marconi-chain-index/src/Marconi/ChainIndex/Experimental/Indexers/MintTokenEvent.hs

```diff
@@ -617,16 +652,26 @@ sortEventsByOrderOfBlockchainAppearance =
     . List.sort
 
 convertEventsMatchingErrorToQueryByAssetIdError
-  :: Core.QueryError (Core.EventsMatchingQuery MintTokenBlockEvents)
-  -> Core.QueryError (QueryByAssetId MintTokenBlockEvents)
-convertEventsMatchingErrorToQueryByAssetIdError = \case
-  Core.NotStoredAnymore -> Core.NotStoredAnymore
-  (Core.IndexerQueryError t) -> Core.IndexerQueryError t
-  (Core.AheadOfLastSync r) -> Core.AheadOfLastSync r
-  (Core.SlotNoBoundsInvalid r) -> Core.SlotNoBoundsInvalid r
+  :: (Monad m)
+  => Core.ListIndexer MintTokenBlockEvents
+  -> Core.QueryError (Core.EventsMatchingQuery MintTokenBlockEvents)
+  -> m (Core.QueryError (QueryByAssetId MintTokenBlockEvents))
+convertEventsMatchingErrorToQueryByAssetIdError indexer = \case
+  Core.NotStoredAnymore -> pure Core.NotStoredAnymore
+  (Core.IndexerQueryError t) -> pure $ Core.IndexerQueryError t
+  (Core.AheadOfLastSync r) -> do
+    res <- case r of
+      Just val -> do
+        r' <- Core.withStability (view Core.point) indexer (pure val)
+        pure (Just r')
```

Comment:
I ended up having to do `traverse (Core.withStability (view Core.point) indexer) $ fmap pure r` which made me realise I should just make `withStability` take a `f (g event)` as its third arg and bind to it in the monadic case, instead of `m (f (g event))`, making this much nicer: `traverse (Core.withStability (view Core.point) indexer) r`